docs/my_documentation.rst

```diff
@@ -0,0 +1,365 @@
+``MULTIPASS_IDENTITY_INFO_KEYS``       Keys used for identification
+``MULTIPASS_LOGIN_SELECTOR_TEMPLATE``  Template with selection of login providers
+``MULTIPASS_LOGIN_FORM_TEMPLATE``      Template with login form
+``MULTIPASS_LOGIN_ENDPOINT``           Endpoint linking to login page
+``MULTIPASS_LOGIN_URLS``               List of login URLs 
+``MULTIPASS_SUCCESS_ENDPOINT``         Endpoint linking to default page after successful login
+``MULTIPASS_FAILURE_MESSAGE``          Message to show after unsuccessfull login
+``MULTIPASS_FAILURE_CATEGORY``         Category of message when flashing after unsuccessful login
+``MULTIPASS_ALL_MATCHING_IDENTITIES``  If true, all matching identities are passed after successful authentication
+``MULTIPASS_REQUIRE_IDENTITY``         If true, ``IdentityRetrievalFailed`` is raised when no matching identities are found, othwerwise empty list is passed
+====================================== =========================================
+
+Configuration example can be found here
+
+< link to config.example.rst>
+
+Registering providers
+---------------------
+
+You can register new provider using the following commands. Remember that your provider class must be a subclass of either ``AuthProvider`` or ``IdentityProvider``:
+
+.. code-block:: python
+
+	class FooProvider(AuthProvider):
+		pass
+
+Then you pass this class to ``register_provider`` with second argument beeing the type name of the provider specified in configuration. Remember that type names must be unique. :
+
+.. code-block:: python
+
+	app = Flask('test')
+	app.config['MULTIPASS_AUTH_PROVIDERS'] = {
+		'test_auth_provider': {'type': 'test'}
+	}
+	multipass = Multipass(app)
+	multipass.register_provider(FooProvider, 'test')
+	
+To instantiate new providers you need to call ``_create_providers`` with key and base class arguments referring to type of providers you registered:
+
+.. code-block:: python
+    	
+ 	multipass._create_providers('AUTH', AuthProvider)
+
+or:
+
+.. code-block:: python
+
+	multipass._create_providers('IDENTITY', IdentityProvider))
+	
+Accessing providers
+---------------------
+	
+Providers can be access through ``auth_providers`` property:
+
+.. code-block:: python
+
+	provider = multipass.auth_providers['test_provider']
+
+or in case of only one provider:
+
+.. code-block:: python
+
+	provider = multipass.single_auth_provider
+
+Processing login
+----------------
+By default, the ``process_login`` method is assigned to the URLs specified in ``MULTIPASS_LOGIN_URLS`` upon initialization of Multipass:
+
+.. code-block:: python
+
+	def _create_login_rule(self):
+		"""Creates the login URL rule if necessary"""
+		endpoint = current_app.config['MULTIPASS_LOGIN_ENDPOINT']
+		rules = current_app.config['MULTIPASS_LOGIN_URLS']
+		if rules is None:
+		    return
+		for rule in rules:
+		    current_app.add_url_rule(rule, endpoint, self.process_login, methods=('GET', 'POST'))
```

Comment:
I wouldn't really specify this code block here. This is an internal method of multipass and people shouldn't need to know this.